components/brave_vpn/browser/connection/brave_vpn_os_connection_api.h

```diff
@@ -29,13 +31,27 @@ class PrefService;
 namespace brave_vpn {
 
 class BraveVPNRegionDataManager;
-struct Hostname;
-
-// Interface for managing OS' vpn connection.
-class BraveVPNOSConnectionAPI
-    : public net::NetworkChangeNotifier::NetworkChangeObserver {
+class ConnectionAPIImpl;
+
+// Interface for managing vpn connection & region data managing.
+//   * BraveVPNRegionDataManager: Manages region data.
+//   * ConnectionAPIImpl: Manages connection state.
+// All client should use this class to manage VPN connection.
+// This class will have proper concrete implemention of ConnectionAPIImpl
+// based on current protocol(ikev2/wireguard).
+class BraveVPNOSConnectionAPI {
```

Comment:
Since this is the interface for Brave to communicate about VPN, there might be a better name for this. Maybe `BraveVPNConnectionManager`. Because it can switch the protocols, it can delegate the calls through to the protocol using `connection_api_impl_`, and also it can install the pre-requisites for example.